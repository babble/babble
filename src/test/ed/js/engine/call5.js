

function foo(){
    return this.a;
}

print( foo.call( { a : 5 } ) );

function foo2( z ){
    return this.a + z;
}

print( foo2.call( { a : 5 } , 3 ) );

function bar(a, b, c){
    print(a);
    print(b);
    print(c);
    return(this.a);
}

print( bar.call( { a: 5}, 1, 2, 3));

print( bar.apply( {a: 5}, [1, 2, 3]));


function bar2(a, b, c){
    print(a == null);
    print(b == null);
    print(c == null);
}

bar2.apply( {a: 5}, [1]);

