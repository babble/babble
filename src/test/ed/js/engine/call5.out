5
8
1
2
3
5
1
2
3
5
false
true
true
